client/src/app/site/agenda/components/agenda-list/agenda-list.component.ts

```diff
@@ -341,4 +342,40 @@ export class AgendaListComponent extends BaseListViewComponent<ViewItem> impleme
             return result;
         }
     }
+
+    /**
+     * Duplicates a single selected item.
+     *
+     * @param item The item to duplicte.
+     */
+    public duplicateSingleTopic(item: ViewItem): void {
+        const topic = item.contentObject.getModel() as Topic;
+        this.topicRepo.create(
+            new Topic({ text: topic.text, title: `${topic.title} - ${this.translate.instant('duplicate')}` })
+        );
+    }
```

Comment:
That should be _either_ more generic:
 - Get the model no matter what it is
 - Get the corresponding repo
 - Call the create function
 - in this case the function name would be "duplicateSingleItem"

**Or** start with an if-else to only allow topics in this function. In the later, throw an error if anything else than a topic was used in this function.